src/event_parser.rs

```diff
@@ -33,7 +33,7 @@ impl Event {
         self.id = value.to_vec();
     }
 
-    fn to_dispatch_event(self) -> Option<Self> {
+    fn format_for_dispatch(self) -> Option<Self> {
```

Comment:
Clippy doesn't like to_ methods that aren't &self, and I really do want to take ownership here.